src/aiovodafone/api.py

```diff
@@ -490,6 +534,118 @@ async def logout(self) -> None:
+        """Trigger a ping diagnostic request to the router.
+
+        Args:
+        ----
+            ip_address (str): The target IP address to ping.
+            count (int): Number of ping requests to send (default: 1).
+            ping_size (int): The size of the ping packet (default: 56 bytes).
+            ping_interval (int): Interval between ping requests in milliseconds
+                (default: 1000 ms).
+            retries (int): Number of times to retry if results are not ready
+                (default: 15).
+
+        Returns:
+        -------
+            dict: The ping results.
+
+        """
+        check_url = "ping"
+        data_url = "ping_res"
+        key = "ping_result"
+        payload = {
+            "ipaddress": ip_address,
+            "count": count,
+            "pingsize": ping_size,
+            "pingintervalin": ping_interval,
+        }
+
+        return await self._trigger_diagnostic_call(
+            check_url, data_url, payload, key, retries
+        )
+
+    async def traceroute(
+        self,
+        ip_address: str,
+        count: int = 30,
+        ip_type: str = "Ipv4",
+        retries: int = 15,
+    ) -> dict[str, Any]:
+        """Trigger a traceroute diagnostic request to the router.
+
+        Args:
+        ----
+            ip_address (str): The target IP address for the traceroute.
+            count (int): Maximum number of hops (default: 30).
+            ip_type (str): IP address type, either "Ipv4" or "Ipv6"
+                (default: "Ipv4").
+            retries (int): Number of times to retry if results are not ready
+                (default: 15).
+
+        Returns:
+        -------
+            dict: The traceroute results.
+
+        """
+        check_url = "traceroute"
+        data_url = "traceroute_res"
+        key = "traceroute_result"
+        payload = {
+            "traceroute_ip": ip_address,
+            "count_tr": str(count),
+            "ipaddresstype": ip_type,
+        }
+
+        return await self._trigger_diagnostic_call(
+            check_url, data_url, payload, key, retries
+        )
+
+    async def dns_resolve(
+        self,
+        hostname: str,
+        dns_server: str = "1.1.1.1",
+        record_type: str = "A",
+        retries: int = 15,
+    ) -> dict[str, Any]:
+        """Trigger a traceroute diagnostic request to the router.
+
+        Args:
+        ----
+            hostname (str): The hostname to resolve.
+            dns_server (str): The DNS server to query (default: 1.1.1.1).
+            record_type (str): DNS record type (default: "A").
+            retries (int): Number of times to retry if results are not ready
+                (default: 15).
+
+        Returns:
+        -------
+            dict: The dns_resolve results.
+
+        """
+        check_url = "tracedns"
+        data_url = "traceDns_res"
+        # The result field is named "traceroute_result"
+        # this is not a typo
+        key = "traceroute_result"
```

Comment:
Should we log something here?
Cause this typo on the API just owned me, had to double-check my test code, hehe.